homeassistant/components/jvc_projector/config_flow.py

```diff
@@ -32,9 +32,11 @@ async def async_step_user(
         errors = {}
 
         if user_input is not None:
+            name = user_input[CONF_NAME]
             host = user_input[CONF_HOST]
             port = user_input[CONF_PORT]
             password = user_input.get(CONF_PASSWORD)
+            readonly = user_input[CONF_READONLY]
```

Comment:
Why do we change the config flow when we are only adding a binary sensor?

Reply:
We should not add these things at all in the config flow. They are not needed.

Reply:
@joostlek as explained in the description addying the Binary Sensor I need to have the name for the different Binary Sensors for the different projectors to be able to distinguish them.
So Name is needed to create the Binary Sensor correctly

Adding the Binary Sensor the "remote" is not mandatory anymore so is correct to add the flag to remove it. As per guidelines.
So the two options are needed to allow to create the Binary Sensor alone

Reply:
@MartinHjelmare as explained to @joostlek they are needed to avoid useless creation of remote and to avoid issue of creating several devices/entities all with the same name.

Reply:
But you can already distinguish devices by renaming them. This is the way we do this at Home Assistant and this is in line with other integrations. Only helpers are allowed to have a settable name.

About the readonly, if you want to add that, sure. But not in this PR. Create a separate one after this with valid rationale and we'll review it.

Reply:
No. Please allow the device name to use the config entry title name, which it will do by default. Then follow our entity naming guidelines.

https://developers.home-assistant.io/docs/core/entity#entity-naming

Reply:
Users can disable the remote entity if they don't need it.

Reply:
@joostlek ok, but you're making my life very complex. I already had to split the PR in 2, now for 1 line of code I have to split again... a little perplex :)

Reply:
@MartinHjelmare sorry but I do not follow you. The name is not hard coded, is asked to the user. The default value "JVC Projector" is the name that is provided by the device as default. The name is not of the entity but of the device.
In fact the entity naming convention is followed: 

> The entity is a member of a device and entity.name is None: friendly_name = f"{device.name}"

This is ok for both Remote and Binary Sensor.

Which is the desired way to change the Device name at creation time? not the entity

Reply:
@MartinHjelmare can you look at the above screenshots, aren't they ok? aren't they following the guideline?
thank you for your help

Reply:
At Home Assistant we made the decision to only allow setting a name for helpers. Setting a name is a common use case for every device and yes, maybe it's a bit messy, but something worse than having a suboptimal way of doing something is if its inconsistent across integrations. (One via the config flow, another via a panel, another one via device). Thus we chose to do it this way to keep it consistent across the board.

Yes there are still integrations that do this, but we are slowly phasing them out and new integrations or features wanting to utilize this are avoided.

Reply:
@joostlek thank you for your patience!
ok, understood... I don't like it :) heheheh but I understood the point. I'll adhere to the new standard

For the disabled can I leave it in or I should split the PR again in another 2?

Reply:
I am going to be upfront, I don't think the readonly change is a good change as well. Like Martin said, people can disable the remote entity if they don't want it. I don't see any reason why someone would want to disable it.

The piece of documentation you referenced to about disabling entities, has Unifi as example. The fact with unifi is that it creates devices and entities on the fly. When someone new connects to your network. Maybe you have a separate open guest network which causes a lot of devices to be added. In those cases I can understand that you want a config flow option to disable wireless entities to be created. (And disabling devices and entities would be an ongoing thing since they get added again)

For such a minor entity, I think this is overkill.

The reason I said I want to have it in a separate PR is that we have a separate place to discuss this so it doesn't block this PR. That's also one of the reasons why we like small PRs.

Reply:
@joostlek great. Will do... in the end this is my first PR!!!

Reply:
Technically its your second ;)

Reply:
But let's remove the thing and we can start looking at it

Reply:
@joostlek heheheh you're right!

I should have removed everything and now should be clean. Just the binary. Can you check if is ok? (or if I've done a mess with github)

Reply:
You haven't pushed yet